### DISCUSSION TITLE
While using appDir, in Client Component, importing a lib still throw "window is not defined" on the server side

### DISCUSSION THREAD

**Discussion by YoannBuzenet** [vercel/next.js#46180] [Help] [answered]
Hello,

I'm working with a library I made and implemented it in next 13 with `appDir` structure. I'm using it only in Client component, because as it is bundled, it uses `window.FormData`. But even there, it is still throwing server-side : `window is not defined`. I thought the code would not run on the server ?

I wonder if I'm using it wrongly, or if I bundled my app wrongly. Can we use window in a bundled library in Client Components ?

**Comment by icyJoseph** [comment, marked as answer]
"Client Components", is a bit of a misleading name. These are rendered server side too.

Imagine if you didn't ship the HTML of client components, it would impact SEO greatly!  In other words, to collect their HTML they are "React rendered" during SSR too.

"Client Components", are the components of old, "Server Components" are the special ones, they run only on the server, and never leak their code to the browser.

So you can't do unguarded window/DOM access in client components, just like you couldn't do it with pages. That hasn't changed.

You need to check for window presence, or do it within useEffects.

**Comment by YoannBuzenet** [comment]
Thank you @icyJoseph, I'll be watching that.

**Comment by YoannBuzenet** [comment]
It was coming from axios, which has `form-data` as a dependency, which adds breaking code using window in the bundle.
Using `fetch()` solved the problem.

**Reply by icyJoseph** [reply]
Less JavaScript shipped to the client too :) - strange though, why can't axios tell it is running server side (axios supports Node.js out of the box) - the plot thickens...
